pkg/types/hashedrekord/hashedrekord.go

```diff
@@ -58,32 +87,34 @@ func ToLogEntry(hr *pb.HashedRekordRequestV0_0_2, algorithmRegistry *signature.A
 	if err != nil {
 		return nil, fmt.Errorf("parsing verifier: %w", err)
 	}
+	return v, nil
+}
 
-	algDetails, err := signature.GetAlgorithmDetails(hr.Signature.Verifier.KeyDetails)
+func verifySupportedAlgorithm(keyDetails v1.PublicKeyDetails, v verifier.Verifier, algorithmRegistry *signature.AlgorithmRegistryConfig) (signature.AlgorithmDetails, error) {
```

Comment:
nit: add a function comment, since this isn't just verifying but also returning the alg details